queue/retry.go

```diff
@@ -0,0 +1,256 @@
+package queue
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/recovery"
+	"github.com/pkg/errors"
+)
+
+type retryHandler struct {
+	// kim: TODO: should add a RetryableQueue interface with an extra PutRetry()
+	// method. That way, it's optional to implement the retryability features.
+	queue         amboy.RetryableQueue
+	opts          amboy.RetryHandlerOptions
+	pending       map[string]amboy.RetryableJob
+	started       bool
+	wg            sync.WaitGroup
+	mu            sync.RWMutex
+	cancelWorkers context.CancelFunc
+}
+
+func newRetryHandler(q amboy.RetryableQueue, opts amboy.RetryHandlerOptions) (amboy.RetryHandler, error) {
```

Comment:
I don't think it has to be since it's only being used for the remoteQueue right now, which is not exported. The `RetryableQueue` is only visible in the top level interfaces for documentation purposes to compare against `RetryableJob` (which devs will have to use) and I doubt anyone will write their own queue implementation outside of this.